prediction_market_agent/agents/blockchain_coding_agent/functions.py

```diff
@@ -0,0 +1,143 @@
+import typing as t
+
+import requests
+import tenacity
+from autogen import ConversableAgent, register_function
+from tavily import TavilyClient
+
+from prediction_market_agent.utils import APIKeys
+
+
+@tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_fixed(1))
+def tavily_search(
```

Comment:
Why not use https://github.com/gnosis/prediction-market-agent-tooling/blob/c6ae2044b41d0d87b81ed912cce800936fff3cb5/prediction_market_agent_tooling/tools/tavily/tavily_search.py#L18 ?

Reply:
implemented